app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,44 @@
+class InvitationsController < ApplicationController
+
+  before_action :require_user, only: [:new, :create]
+
+  def new
+    @invitation = Invitation.new
+    @invitation.user = current_user
+  end
+
+  def create
+    params = invitation_params
```

Comment:
The performance gains are extremely minimal. Ruby is about easy to read and understand code. You are giving that up here in my opinion.